source/extensions/filters/http/transformation/inja_transformer.h

```diff
@@ -86,11 +87,23 @@ class Extractor : Logger::Loggable<Logger::Id::filter> {
 private:
   absl::string_view extractValue(Http::StreamFilterCallbacks &callbacks,
                                  absl::string_view value) const;
+  absl::string_view replaceValue(Http::StreamFilterCallbacks &callbacks,
+                                 absl::string_view value) const;
+  absl::string_view replaceIndividualValue(Http::StreamFilterCallbacks &callbacks,
+                                           absl::string_view value) const;
+  absl::string_view replaceAllValues(Http::StreamFilterCallbacks &callbacks,
+                                     absl::string_view value) const;
 
   const Http::LowerCaseString headername_;
   const bool body_;
   const unsigned int group_;
   const std::regex extract_regex_;
+  const bool has_replacement_text_;
+  const std::string replacement_text_;
```

Comment:
`has_replacement_text_` and `replacement_text_` are each derived from the `google.protobuf.stringValue` API field  `replacement_text`. We could have a single `google::protobuf::stringValue` field on the extractor that is 1:1 with the API field, but I think having a separate bool/string is cleaner. That being said, I'm open to changing this if reviewers have strong feelings

Reply:
Updated to use a std::optional here: https://github.com/solo-io/envoy-gloo/pull/301/commits/71000c6b9712d53722b6cb0cb4ad2549a2444f19